src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs

```diff
@@ -96,10 +97,16 @@ private static IDiagnoser[] LoadClassic()
 
         private static Assembly LoadDiagnosticsAssembly(Assembly benchmarkDotNetCoreAssembly)
         {
+            // We couldn't use Assembly.GetEntryAssembly()?.GetReferencedAssemblies() because
             // it not enough to just install NuGet to be "referenced", the project has to consume the dll for real to be on the referenced assembly list
-            var referencedAssemblyName = Assembly.GetEntryAssembly()?.GetReferencedAssemblies().SingleOrDefault(name => name.Name == DiagnosticAssemblyName);
-            if (referencedAssemblyName != default(AssemblyName))
-                return Assembly.Load(referencedAssemblyName);
+            // So we use DependencyContext that reads *.deps.json file that contains all referenced assembly. 
+            var inCompileLibraries = DependencyContext.Default.CompileLibraries.Any(l => l.Name.Equals(DiagnosticAssemblyName, StringComparison.OrdinalIgnoreCase));
+            var inRuntimeLibraries = DependencyContext.Default.RuntimeLibraries.Any(l => l.Name.Equals(DiagnosticAssemblyName, StringComparison.OrdinalIgnoreCase));
+
+            if (inCompileLibraries || inRuntimeLibraries)
+            {
+                return Assembly.Load(new AssemblyName(DiagnosticAssemblyName));
```

Comment:
I would assume that the `DependencyContext` exposes some method to load the library without providing the name. Here you just check that it's on the list and if it is you load it by name. How `Assembly.Load` will know about the `.deps` file here?

@eerhardt how should we use `DependencyContext` to load an assembly?

Reply:
I found this way there https://github.com/dotnet/corefx/issues/11639#issuecomment-300565009.
I created project on the side and this solution works.

I don't know how `Assembly.Load` works. I'll try to check.

Reply:
`DependencyContext` allows you to get the information about a dependency, but it doesn't have assembly loading capabilities. You still use the normal `Assembly.Load` APIs to load the assembly.

The reason I suggested using `DependencyContext` is because I assumed the main thing you need/want here is you want to get the AssemblyVersion of the referenced `BenchmarkDotNet.Diagnostics.Windows.dll`, right?  (That information can be found [on this API](https://github.com/dotnet/core-setup/blob/bb31dc3b52e38ca4e679f741cd0263dc239b926f/src/managed/Microsoft.Extensions.DependencyModel/RuntimeFile.cs#L24)).

If it works to just `Assembly.Load(new AssemblyName(DiagnosticAssemblyName));`, then you don't need the `DependencyContext` nor the bit about `Assembly.GetEntryAssembly()?.GetReferencedAssemblies()` at all.  Just call `Assembly.Load` passing in the .dll name.

Reply:
Calling `Assembly.Load` is enough for .net core and full framework. 
For full framework, it works even if instead of referencing a library, I copy the dll to the bin directory.